### PR TITLE
Add Coolcodename

### PR DESCRIPTION
**Pull request by Karthik99999**

### CODE REVIEW

File: data/mods/ssb/abilities.ts [hunk 1]
```diff
@@ -100,6 +100,21 @@ export const Abilities: {[k: string]: ModdedAbilityData} = {
...
 	},
 
+	// Coolcodename
+	firewall: {
+		shortDesc: "Burns opponents that attempt to use status moves on this Pokemon; Status move immunity.",
+		name: "Firewall",
+		onTryHit(target, source, move) {
+			if (move.category === 'Status' && target !== source) {
+				if (!source.trySetStatus('brn', target)) {
+					this.add('-immune', target, '[from] ability: Good as Gold');
```
Comment (HisuianZoroark) [inline data/mods/ssb/abilities.ts:110]:
```suggestion
				this.add('-immune', target, '[from] ability: Firewall');
				source.trySetStatus('brn', target));
```

File: data/mods/ssb/conditions.ts [hunk 2]
```diff
@@ -89,6 +89,17 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
+		onStart() {
+			this.add(`c:|${getName('deftinwolf')}|LFGI (Player 1)`);
+		},
+		onSwitchOut() {
+			this.add(`c:|${getName('deftinwolf')}|right, i forgot i have a skill issue`);
+		},
+		onFaint() {
+			this.add(`c:|${getName('deftinwolf')}|mb LOL`);
+		},
+	},
```
Comment (HisuianZoroark) [inline data/mods/ssb/conditions.ts:102]: >:(

Comment (HisuianZoroark) [inline data/mods/ssb/conditions.ts:102]: player 1 should should probably the user, also deftinwolf

File: data/mods/ssb/random-teams.ts [hunk 3]
```diff
@@ -60,6 +60,12 @@ export const ssbSets: SSBSets = {
 		signatureMove: "Baker's Douze Off",
 		evs: {hp: 252, def: 252, spd: 4}, nature: 'Impish', teraType: 'Fairy',
 	},
+	Coolcodename: {
+		species: 'Victini', ability: 'Firewall', item: 'Heavy-Duty Boots', gender: 'M',
+		moves: ['Searing Shot', 'Psychic', 'Dazzling Gleam'],
+		signatureMove: "Haxer's Will",
+		evs: {spa: 252, spd: 4, spe: 252}, nature: 'Timid', teraType: 'Fairy',
```
Comment (HisuianZoroark) [inline data/mods/ssb/random-teams.ts:67]: should be the randbats shiny odds here (1/1024?)

File: data/mods/ssb/abilities.ts (hunk 1 above)
Comment (Karthik99999) [reply data/mods/ssb/abilities.ts:110]: Usually if there's another effect the immunity message is not shown. Will fix me leaving good as gold on there though

**Comment by HisuianZoroark** [comment]
looks good!

**Pull request merged**